.travis.yml

```diff
@@ -21,44 +21,100 @@ compiler:
   - gcc
```

Comment:
Can we get away with removing this compiler directive from the build matrix? It looks like we're manually installing the gcc6/clang packages and using env-vars to determine which to use. Helps us reduce the # of excludes.